docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+<https://docs.getdbt.com/docs/build/data-tests>`_, and the `dbt_expectations
+<https://github.com/metaplane/dbt-expectations>`_ package provides many more. We also
+define our own `custom data tests
+<https://docs.getdbt.com/best-practices/writing-custom-generic-tests>`_.
+
+.. note:: Why use dbt and SQL instead of Python?
+
+   Modern analytical query engines are extremely fast and memory efficient. By using dbt
+   with `DuckDB <https://duckdb.org/>`_ to query the PUDL `Parquet
+   <https://parquet.apache.org/>`_ outputs we're able to run thousands of validations
+   across hundreds of tables with billions of rows in a minute, instead of the 2-3 hours
+   it used to take our much less extensive validation tests to run. Plus we get to learn
+   SQL.
+
+--------------------------------------------------------------------------------
+Data validation guidelines
+--------------------------------------------------------------------------------
+
+.. todo::
+
+   We should chat about the minimum level of data validation that we expect for a table
+
+   * row counts
+   * no entirely null columns
+   * anything else now?
+
+--------------------------------------------------------------------------------
+Example of typical data validation workflow
+--------------------------------------------------------------------------------
+
+* Explain that the dbt tests are looking at whatever Parquet files are in
+  ``$PUDL_OUTPUT``.
+* Need to be aware of the state of your Dagster outputs and how they relate to the
+  branch / code that you're working on. If you have mixed outputs from multiple
+  branches, and you run the full data validation tests, they might fail because of the
+  mismatch.
+* Typical workflow: tweaking an asset, remateralize it in Dagster, re-run the data
+  validations that pertain to just that table.
+* Often also useful to rematerialize the changed table and all of its downstream
+  dependencies, and then run the data validations on all of those downstream
+  dependencies to see if there were any unforeseen consequences.
+
+--------------------------------------------------------------------------------
+Running the data validation tests
+--------------------------------------------------------------------------------
+
+Setting up dbt
+~~~~~~~~~~~~~~
+
+The dbt directory contains the PUDL dbt project which manages our `data tests
+<https://docs.getdbt.com/docs/build/data-tests>`_. To run dbt you'll need to have the
+``pudl-dev`` conda environment activated (see :doc:`dev_setup`). Before running the
+tests you'll want to run the following from within the ``dbt/`` directory in your pudl
+repo:
+
+.. code-block:: bash
+
+    # Install additional dbt-specific dependencies in the dbt project
+    dbt deps
+    # Materializes dbt-specific "seed" tables from CSVs in the seeds/ directory
+    # In our case, these contain the expected row counts for PUDL tables
+    dbt seed
```

Comment:
These were both in the old docs, but I'm not sure if they're truly required -- running `dbt build` seems to rematerialize the seeds tables.